evap/staff/tests/test_views.py

```diff
@@ -185,64 +187,137 @@ def setUpTestData(cls):
+        error_string = (
+            'Multiple users match the email testuser1@institution.example.com:'
+            + '<br />Elisabeth Fröhlich (testuser1@institution.example.com)'
+            + '<br />Tony Kuchenbuch (testuser1@internal.example.com)'
+        )
+        button_substring = 'value="bulk_update"'
+
+        expected_users = set(UserProfile.objects.all())
 
-    def test_deletes_users(self):
-        baker.make(UserProfile, username='testuser1')
-        baker.make(UserProfile, username='testuser2')
+        page = self.app.get(self.url, user='manager')
+        form = page.forms["user-bulk-update-form"]
+        form["username_file"] = (self.filename,)
+        response = form.submit(name="operation", value="test")
+
+        self.assertIn(button_substring, response)
+        self.assertNotIn(error_string, response)
+        self.assertEqual(set(UserProfile.objects.all()), expected_users)
+
+        new_user = baker.make(UserProfile, first_name="Tony", last_name="Kuchenbuch", email='testuser1@internal.example.com')
+        expected_users.add(new_user)
+
+        page = self.app.get(self.url, user='manager')
+        form = page.forms["user-bulk-update-form"]
+        form["username_file"] = (self.filename,)
+        response = form.submit(name="operation", value="test")
+
+        self.assertNotIn(button_substring, response)
+        self.assertIn(error_string, response)
+        self.assertEqual(set(UserProfile.objects.all()), expected_users)
+
+    @override_settings(INSTITUTION_EMAIL_DOMAINS=["institution.example.com", "internal.example.com"])
+    def test_handles_users(self):
+        baker.make(UserProfile, email='testuser1@institution.example.com')
+        baker.make(UserProfile, email='testuser2@institution.example.com')
+        baker.make(UserProfile, email='testupdate@institution.example.com')
         contribution1 = baker.make(Contribution)
         semester = baker.make(Semester, participations_are_archived=True)
-        evaluation = baker.make(Evaluation, course=baker.make(Course, semester=semester), _participant_count=0, _voter_count=0)
+        evaluation = baker.make(Evaluation,
+            course=baker.make(Course, semester=semester), _participant_count=0, _voter_count=0
+        )
         contribution2 = baker.make(Contribution, evaluation=evaluation)
-        baker.make(UserProfile, username='contributor1', contributions=[contribution1])
-        baker.make(UserProfile, username='contributor2', contributions=[contribution2])
+        baker.make(UserProfile, email='contributor1@institution.example.com', contributions=[contribution1])
+        baker.make(UserProfile, email='contributor2@institution.example.com', contributions=[contribution2])
 
-        page = self.app.get(self.url, user='manager')
-        form = page.forms["user-bulk-delete-form"]
+        expected_users = set(UserProfile.objects.all())
+        expected_users.discard(UserProfile.objects.get(email='testuser2@institution.example.com'))
```

Comment:
What about `UserProfile.objects.exclude`? Or did you want to raise an error if `testuser2@institutation.example.com` does not exists?